### PR TITLE
Functor branch

### PR DESCRIPTION
**Pull request by arestrepogiraldo2200**
### Description
I created a new entry in the cpp/concepts/ directory called functors (not functions). Inside the cpp/concepts/functors there is a markdown functors.md with the topics and format according to the template suggested. 

### Type of Change

- Adding a new entry

### Checklist

- [x] All writings are my own.
- [x] My entry follows the Codecademy Docs style guide.
- [x] My changes generate no new warnings.
- [x] I have performed a self-review of my own writing and code.
- [x] I have checked my entry and corrected any misspellings.
- [x] I have made corresponding changes to the documentation if needed.
- [x] I have confirmed my changes are not being pushed from my forked `main` branch.
- [x] I have confirmed that I'm pushing from a new branch named after the changes I'm making.
- [x] Under "Development" on the right, I have linked any issues that are relevant to this PR (write "Closes #<issue number> in the "Description" above).

### CODE REVIEW

File: content/cpp/concepts/functors/functors.md [hunk 1]
```diff
@@ -0,0 +1,76 @@
+
```
Comment (SSwiniarski) [inline content/cpp/concepts/functors/functors.md:1]: Remove the line break before the metadata.

File: content/cpp/concepts/functors/functors.md [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+---
+Title: 'Functors'
+Description: 'A functor is an object or struct that can be called like a function by overloading the function-call operator ().'
+
```
Comment (SSwiniarski) [inline content/cpp/concepts/functors/functors.md:5]: Remove the extra line breaks between the metadata sections.

File: content/cpp/concepts/functors/functors.md [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+  - 'Classes'
+  - 'Objects'
+  - 'OOP'
+  - 'Functions'
+
+CatalogContent:
+  - 'learn-c-plus-plus'
+  - 'paths/computer-science'
+---
+
```
Comment (SSwiniarski) [inline content/cpp/concepts/functors/functors.md:21]: Remove extra line break after metadata.

File: content/cpp/concepts/functors/functors.md [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+
+A **functor** is an object or struct that can be called like a function by overloading the function-call operator (). Public access must be granted to the overloading of () operator in order to be used as intended. Functors can simplify tasks and improve efficiency in many cases.
```
Comment (SSwiniarski) [inline content/cpp/concepts/functors/functors.md:23]:
```suggestion
A **functor** is an object or struct that can be called like a function by overloading the function-call operator `()`. Public access must be granted to the overloading of the `()` operator in order to be used as intended. Functors can simplify tasks and improve efficiency in many cases.
```

File: content/cpp/concepts/functors/functors.md [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+
+```cpp
+class MyClass{
+  public:
+    type operator()(...) {
+      // function body
+    }
+};
+```
+where `type` is the [data type](https://www.codecademy.com/resources/docs/cpp/data-types) to be returned and `(...)`are the arguments to be given for the execution. Then, to use the functor just create an instance of the class and when needed call it passing the arguments required as
```
Comment (SSwiniarski) [inline content/cpp/concepts/functors/functors.md:37]:
```suggestion

Where `type` is the [data type](https://www.codecademy.com/resources/docs/cpp/data-types) to be returned and `(...)`are the arguments to be given for the execution. Then, to use the functor just create an instance of the class and when needed call it passing the arguments required as
```

File: content/cpp/concepts/functors/functors.md [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+int main(){
+
+    Hello salute;
+    salute(3);
+
+    return 0;
+}
+```
+
+returns
```
Comment (SSwiniarski) [inline content/cpp/concepts/functors/functors.md:70]:
```suggestion
Results in the following output:
```

File: content/cpp/concepts/functors/functors.md [hunk 7]
```diff
@@ -0,0 +1,76 @@
...
+where `type` is the [data type](https://www.codecademy.com/resources/docs/cpp/data-types) to be returned and `(...)`are the arguments to be given for the execution. Then, to use the functor just create an instance of the class and when needed call it passing the arguments required as
+
+```cpp
+ MyClass myclass;
+ myclass(...);
+```
+
+## Example
+
+The following code
```
Comment (SSwiniarski) [inline content/cpp/concepts/functors/functors.md:46]:
```suggestion
The following code:
```

**Review by SSwiniarski — CHANGES_REQUESTED** [review]
@arestrepogiraldo2200 I did a first review and left a few comments.

**Comment by arestrepogiraldo2200** [comment]
Hello @SSwiniarski, I just adopted the corrections you suggested. Merry christmas.

File: content/cpp/concepts/functors/functors.md [hunk 8]
```diff
@@ -0,0 +1,72 @@
...
+  - 'paths/computer-science'
+---
+
+A **functor** is an object or struct that can be called like a function by overloading the function-call operator `()`. Public access must be granted to the overloading of the `()` operator in order to be used as intended. Functors can simplify tasks and improve efficiency in many cases.
+
+## Syntax
+
+For an object to be a functor, the class body must contain the following lines 
+
+```cpp
```
Comment (SSwiniarski) [inline content/cpp/concepts/functors/functors.md:24]: change this to a ```pseudo block.

File: content/cpp/concepts/functors/functors.md [hunk 9]
```diff
@@ -0,0 +1,72 @@
...
+  public:
+    type operator()(...) {
+      // function body
+    }
+};
+```
+
+Where `type` is the [data type](https://www.codecademy.com/resources/docs/cpp/data-types) to be returned and `(...)`are the arguments to be given for the execution. Then, to use the functor just create an instance of the class and when needed call it passing the arguments required as
+
+```cpp
```
Comment (SSwiniarski) [inline content/cpp/concepts/functors/functors.md:35]: Make this a ```pseudo block

**Review by SSwiniarski — CHANGES_REQUESTED** [review]
@arestrepogiraldo2200 I had just a couple more comments.

**Comment by arestrepogiraldo2200** [comment]
Done

File: content/cpp/concepts/functors/functors.md [hunk 10]
```diff
@@ -0,0 +1,72 @@
...
+  - 'paths/computer-science'
+---
+
+A **functor** is an object or struct that can be called like a function by overloading the function-call operator `()`. Public access must be granted to the overloading of the `()` operator in order to be used as intended. Functors can simplify tasks and improve efficiency in many cases.
+
+## Syntax
+
+For an object to be a functor, the class body must contain the following lines 
+
+```pseudo block
```
Comment (SSwiniarski) [inline content/cpp/concepts/functors/functors.md:24]: just ```pseudo

File: content/cpp/concepts/functors/functors.md [hunk 11]
```diff
@@ -0,0 +1,72 @@
...
+  public:
+    type operator()(...) {
+      // function body
+    }
+};
+```
+
+Where `type` is the [data type](https://www.codecademy.com/resources/docs/cpp/data-types) to be returned and `(...)`are the arguments to be given for the execution. Then, to use the functor just create an instance of the class and when needed call it passing the arguments required as
+
+```pseudo block
```
Comment (SSwiniarski) [inline content/cpp/concepts/functors/functors.md:35]: just ```pseudo

**Review by SSwiniarski — CHANGES_REQUESTED** [review]
@arestrepogiraldo2200 just two more things.

**Comment by arestrepogiraldo2200** [comment]
@SSwiniarski correction done

**Review by SSwiniarski — APPROVED** [review]
@arestrepogiraldo2200 Looks good for a second review.

**Comment by arestrepogiraldo2200** [comment]
Hello guys I'm still waiting for feedback. Thanks a lot. :grinning:

File: content/cpp/concepts/functors/functors.md [hunk 12]
```diff
@@ -0,0 +1,72 @@
...
+The following code:
+
+```cpp
+#include<iostream>
+using namespace std;
+
+class Hello{
+
+    public:
+        void operator()(int times){
```
Comment (yangc95) [inline content/cpp/concepts/functors/functors.md:51]: It may be good to include something about this operator function.

**Review by yangc95 — COMMENTED** [review]
@arestrepogiraldo2200 thanks for following up! Just went over this entry and added some comments/suggestions. Tag me when you've resolved these 🙂 or if you want to discuss something

File: content/cpp/concepts/functors/functors.md [hunk 13]
```diff
@@ -0,0 +1,72 @@
+---
+Title: 'Functors'
+Description: 'A functor is an object or struct that can be called like a function by overloading the function-call operator ().'
```
[the remark this answers is not in the document]
Comment (arestrepogiraldo2200) [reply content/cpp/concepts/functors/functors.md:3]: A functor can be created from either a `class` or a `struct`. I based mainly in the following:

[GeeksForGeeks](https://www.geeksforgeeks.org/functors-in-cpp/)
[Programiz](https://www.programiz.com/cpp-programming/functors)
[StackOverflow](https://stackoverflow.com/questions/356950/what-are-c-functors-and-their-uses)

I just wrote a little example with a struct:

```cpp
#include<iostream>
using namespace std;

int main(){

   struct functor{
      int x; 
      void operator ()(int y){ 
          x += y;
      }
   };
   
   functor example;               // creates struct instance
   example.x = 3;                  // sets x to 3
   example(4);                      // runs operator overload with instance "example"
   cout<<"Now x is: "<<example.x<<endl;        // prints "Now x is 7"

   return 0;
}
```

[the remark this answers is not in the document]
Comment (yangc95) [reply content/cpp/concepts/functors/functors.md:3]: Thanks for sharing! Maybe it's a good idea to have both class and structure mentioned

**Comment by arestrepogiraldo2200** [comment]
Hello @yangc95 :wave:. I have implemented some of the corrections you indicated. Please tell me what do you think.

**Review by yangc95 — APPROVED** [review]

**Comment by yangc95** [comment]
@arestrepogiraldo2200 very happy to finally get this entry up! Thanks for your resilience throughout this process. Here's the live link: https://www.codecademy.com/resources/docs/cpp/functors

Screen Shot 2023-02-26 at 10 05 42 PM

> It looks as if you're anonymous. If you want to be credited follow the instructions below to add your GitHub username to your Codecademy profile. If not, there is nothing more you need to do.
> 
> To add your GitHub username to your Codecademy profile:
> 
> 1.) Add your GitHub Username to your Codecademy Profile Details, you can get there on your desktop by pressing on your profile picture in the upper right-hand corner and selecting "Profile" in the drop-down box:
> 
> image
> 
> 2.) Then log out and log back into Codecademy using your GitHub login:
> 
> githublogin


cc: @SSwiniarski @KTom101

**Pull request merged**